### ISSUE TITLE
Focus fields web component issues

### ISSUE THREAD

**Issue by rhasselle-oddball** [department-of-veterans-affairs/va.gov-team#62008]
Fields will often not properly focus the correct field (e.g. focus first item) when using web components.

Example: in mock-simple-forms-patterns mockArraySinglePage or mockArrayMultiplePageStart, try adding multiple items, and notice it will not correctly focus the right item, due to not finding inputs in the shadow dom.

will likely need to update ArrayField.jsx focusOnFirstFocusableElement function to also look for va-text-input, va-select, etc... and then check the .shadowRoot if it exists and then querySelector inside of that.

Also test for any other odd focusing issues.

Also test page to page focusing first element.

caution - likely cannot prioritize web component focusing over normal field focusing. likely would need to combine both and find the first. then dig into shadowRoot if it exists to focus element in there.

- [x] Focusing headers on the next page
- [ ] Focusing web components in general
- [x] Focusing web components in arrays

**Comment by PSIEBENLIST** [comment]
**Meeting Decision: Form Focus**

- Pattern Mock Form: Focus should go to the H4 stepper.

- Form Header Prototype: Focus should go to the H1 question (e.g. name and date of birth).

Nice to have:
- Implement scroll behavior before or after page loads for enhanced user experience.

@jeana-adhoc can you confirm requirements?

**Comment by rhasselle-oddball** [comment]
There are three issues that this ticket relates to:
- Focusing headers on the next page
- Focusing web components in general
- Focusing web components in arrays

The array portion of this is blocked by needing to upgrade to Cypress 12. PR: https://github.com/department-of-veterans-affairs/vets-website/pull/25149

**Comment by rhasselle-oddball** [comment]
Fixed focus issue where focus was not on the h2 stepper for web component. 

Closing issue while acknowledging there are still some focus issues:
- First page doesn't seem to focus the correct thing
- Scroll does not always behave appropriately despite focus working
- Sometimes shadow dom does not exist of web components affecting scrolling

**Comment by rhasselle-oddball** [comment]
Moved "Form Header Prototype: Focus should go to the H1 question (e.g. name and date of birth). " to be addressed in #509
